### PR TITLE
[QDP] Optimize run_throughput_pipeline to avoid per-iteration Vec allocations

### LINKED ISSUE #1125: [QDP] Optimize run_throughput_pipeline to avoid per-iteration Vec allocations
### What
`run_throughput_pipeline` currently allocates a new `Vec<f64>` for every batch via generate_batch(...) inside the hot loops (warmup + timed). In throughput benchmarks this adds significant allocator overhead and variance, making results reflect CPU allocation time rather than GPU encode_batch throughput. We should allocate a single batch buffer once and refill it in-place each iteration.

### Location
File: [qdp/qdp-core/src/pipeline_runner.rs](https://github.com/apache/mahout/blob/main/qdp/qdp-core/src/pipeline_runner.rs)
Function: `run_throughput_pipeline` 

### Proposed Change
- Allocate batch_buf: Vec<f64> once outside the loops.
- Replace per-iteration generate_batch(...) with an in-place fill function (e.g. fill_batch_inplace(...)) that writes into &mut [f64].
- Reuse the same batch_buf in both warmup and timed loops.

### PR DESCRIPTION
**Pull request by 0lai0**
### Related Issues

Close [#1125](https://github.com/apache/mahout/issues/1125) 

### Changes

- [x] Bug fix
- [ ] New feature
- [ ] Refactoring
- [ ] Documentation
- [ ] Test
- [ ] CI/CD pipeline
- [ ] Other

### Why

In `run_throughput_pipeline`, each iteration of the warmup and timed loops called `generate_batch(...)`, which allocates a new `Vec<f64>` (`batch_size * vector_len` elements). For high-frequency throughput benchmarks, this creates significant memory allocation pressure and adds allocator overhead to the timing results, making them reflect CPU allocation time rather than pure GPU `encode_batch` throughput.

### How

- Pre-allocate a single batch buffer before entering the loops
- Reuse the buffer in each iteration via a new `fill_batch_inplace(...)` function that writes data in-place
- Refactor `generate_batch` to delegate to `fill_batch_inplace`, eliminating code duplication

## Checklist

- [x] Added or updated unit tests for all changes
- [x] Added or updated documentation for all changes

### CODE REVIEW

File: qdp/qdp-core/src/pipeline_runner.rs [hunk 1]
```diff
@@ -569,3 +583,42 @@ pub fn run_throughput_pipeline(config: &PipelineConfig) -> Result<PipelineRunRes
...
+        let config = PipelineConfig {
+            num_qubits: 5,
+            batch_size: 8,
+            encoding_method: encoding_method.to_string(),
+            seed: Some(123),
+            ..Default::default()
+        };
+
+        let vector_len = vector_len(config.num_qubits, &config.encoding_method);
+        let batch_idx = 7;
```
Comment (ryankert01) [inline qdp/qdp-core/src/pipeline_runner.rs]: we add edge cases like: 0, batch_size-1

Comment (ryankert01) [reply qdp/qdp-core/src/pipeline_runner.rs]: We can also test if adjacent batch is same of not.
```rust
let batch0 = generate_batch(&config, 0, vector_len);
let batch1 = generate_batch(&config, 1, vector_len);
assert_ne!(batch0, batch1);
```

**Review by ryankert01 — COMMENTED** [review]
Implementation looks good, a nit with the tests

File: qdp/qdp-core/src/pipeline_runner.rs (hunk 1 above)
Comment (0lai0) [reply qdp/qdp-core/src/pipeline_runner.rs]: Thanks @ryankert01 for review. I'll update in next commit.

**Review by viiccwen — COMMENTED** [review]
lg!
but we can add more tests like `Seed=None`, `batch_size=1`, `amplitude==AMPLITUDE`, `amplitude sample in [0, 1]`

**Review by ryankert01 — APPROVED** [review]

**Review by rich7420 — APPROVED** [review]

**Comment by rich7420** [comment]
@0lai0 thanks for the patch, @ryankert01 and @viiccwen thanks for the review
